### PR TITLE
web:Fix a potential goroutine leak by channel

### PR DESCRIPTION
**Pull request by lzhfromustc**
***What this PR does***
Fix a potential goroutine blocking problem caused by channel `errch`.

***Description***
https://github.com/prometheus/prometheus/blob/eda5f8ca5f2223620cae36627919bab58c78634f/web/web.go#L592-L603
`errch` is a channel with no buffer.
It has a send operation in a child goroutine, and a receive operation in a select in the parent goroutine.
If the select chooses `case <-ctx.Done()`, then no one can receive from the channel, leaving the child goroutine blocked forever.

***How this problem is fixed***
With 1 buffer given to `errch`, the child goroutine won't be blocked, no matter which case the parent goroutine chooses.
This fix won't change the semantics of the parent goroutine, since the behavior of the receive operation is not changed.

### CODE REVIEW

**Comment by roidelapluie** [comment]
Thanks for your PR. For legal reasons, we require that all commits are signed with a [DCO](https://developercertificate.org/) before we can merge them. See [this blog post](https://www.cncf.io/blog/2016/11/08/cloud-native-software-can-trust/) for considerations around this.

This means that the last line of your commit message should read like:

`Signed-Off-By: Your Name <your@email.address>`

If you are using GitHub through the web interface, it's quickest to close this PR and open a new one with the appropriate line.

If you are using Git on the command line, it is probably quickest to amend and force push. You can do that with
````
git commit --amend --signoff
git push -f $remote $remote_branch_for_pr
````

As always, be careful when force-pushing.

**Comment by songlh** [comment]
Could we merge this pull request? It looks good to me.

**Comment by charlesxsh** [comment]
Hi there, similar to #8964, kindly ask could this PR get merge if nothing wrong?

**Comment by charlesxsh** [comment]
@roidelapluie Hi there, seems like this PR fixed the existing comments. Would you like to merge this PR if everything looks good?

**Review by roidelapluie — APPROVED** [review]
Thanks!

**Pull request merged**
